pandas/core/arrays/sparse/array.py

```diff
@@ -1636,6 +1639,44 @@ def _min_max(self, kind: Literal["min", "max"], skipna: bool) -> Scalar:
         else:
             return na_value_for_dtype(self.dtype.subtype, compat=False)
 
+    def _argmin_argmax(self, kind: Literal["argmin", "argmax"]) -> int:
+
+        values = self._sparse_values
+        mask = np.asarray(isna(self))[self._sparse_index.indices]
+        func = np.argmax if kind == "argmax" else np.argmin
+
+        idx = np.arange(values.shape[0])
+        non_nans = values[~mask]
+        non_nan_idx = idx[~mask]
+
+        _candidate = non_nan_idx[func(non_nans)]
+        candidate = self._sparse_index.indices[_candidate]
+
+        if isna(self.fill_value):
+            return candidate
+        if kind == "argmin" and self[candidate] < self.fill_value:
+            return candidate
+        if kind == "argmax" and self[candidate] > self.fill_value:
+            return candidate
+        _loc = self._first_fill_value_loc()
```

Comment:
As you mentioned below, `_first_fill_value_loc` appears to be named confusingly and does more work than needed (involves a sort as well). If you just want to see whether or not the fill value exists, could use `ngaps` or `npoints` on `self.sp_index` (for example as done in `_min_max` or `unique`)

Reply:
No, actually the name is fine, but the doc and implementation is wrong. I do need to find the first position of the fill value in argmax/argmin instead of just checking the existence.

Reply:
But for the unique function, you are right. We can use O(1) instead O(log(n)) to check the existence.

Reply:
I can give you the example:
```
>>> arr = SparseArray([1, 2, 0, 0], fill_value=0)
>>> arr.argmin()
2
```
The candidate is 1 but fill_value is smaller, thus we return the result from _first_fill_value_loc().

Reply:
Ok thanks for explaining, makes sense. Based on code in `SparseIndex`, for example https://github.com/pandas-dev/pandas/blob/9f43ec6a98580d8f1c74d068afb31ec902c63e8c/pandas/_libs/sparse.pyx#L75, we should be guaranteed monotonically increasing indices. So is the first fill value location just `self.sp_index.indices[0]`?

Reply:
`self.sp_index.indices[0]` is actually the first non-fill_value position, not the fill_value.